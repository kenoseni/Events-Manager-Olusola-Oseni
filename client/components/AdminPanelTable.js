import React, {Component} from 'react';

class AdminPanelTable extends Component {
  constructor(props) {
    super(props);
    this.upgradeUserToAdmin = this.upgradeUserToAdmin.bind(this);
  }
  
  upgradeUserToAdmin (e) {
    e.preventDefault();
    this.props.upgradeUserToAdmin(this.props.user.id);
  }
  render () {
    const {user, i} = this.props
    const {events, centers} = this.props.user
    return (
      <tr>
        <th scope="row">{i}</th>
        <td>{user.id}</td>
        <td>{user.firstname}</td>
        <td>{user.lastname}</td>
        <td>{user.isAdmin.toString()}</td>
        <td>{user.role}</td>
        <td>{user.email}</td>
        <td><button onClick={this.upgradeUserToAdmin} type="button" className="btn btn-primary btn-sm my-0">UPGRADE</button></td>
      </tr>
    )
  }
}
export default AdminPanelTable;
