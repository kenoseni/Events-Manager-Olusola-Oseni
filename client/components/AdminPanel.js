import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import AdminPanelTable from './AdminPanelTable';
import Pages from './Pages';

class AdminPanel extends Component {
  render() {
    const { users, match, history, count, limit, userDetails } = this.props;
    return (
      <div className="container">
        <div className="row" id="">
          <div className="col-xs-12 col-md-0" />
          <div className="col-xs-12 col-md-12" id="etable">
            <div className="card w-40">
              <div className="card-body">
                <div className="table-wrapper">
                  <h3 id="appusers">Users</h3>
                  {/*<!--Table-->*/}
                  <table className="table table-hover table-responsive">
                    <thead className="blue-grey text-white lighten-1">
                      <tr>
                        <th>#</th>
                        <th className="th-lg">ID</th>
                        <th className="th-lg">FIRST NAME</th>
                        <th className="th-lg">LAST NAME</th>
                        <th className="th-lg">ISADMIN</th>
                        <th className="th-lg">ROLE</th>
                        <th className="th-lg">EMAIL</th>
                        {userDetails.userrole === 'superadmin' && (
                          <th className="th-lg">ACTION</th>
                        )}
                        {userDetails.userrole === 'admin' && (
                          <th className="th-lg" />
                        )}
                      </tr>
                    </thead>
                    <tbody id="adminpaneltable">
                      {users
                        ? users.map((user, i) => (
                            <AdminPanelTable
                              {...this.props}
                              key={i}
                              i={i + 1}
                              user={user}
                            />
                          ))
                        : null}
                    </tbody>
                  </table>
                  {/*<!--Table-->*/}
                  <div>
                    <div
                      style={{
                        display: 'flex',
                        justifyContent: 'space-between'
                      }}
                    >
                      <span className="count">
                        {users
                          ? `1 - ${users.length} of ${users.length}`
                          : null}
                      </span>
                      <span>
                        {count !== undefined && (
                          <Pages
                            count={count}
                            history={history}
                            limit={limit}
                          />
                        )}
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="col-xs-12 col-md-0" />
        </div>
      </div>
    );
  }
}

export default AdminPanel;
