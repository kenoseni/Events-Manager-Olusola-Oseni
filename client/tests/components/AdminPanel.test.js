import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Connected, { AdminPanels } from '../../components/AdminPanels';
import AdminPanel from '../../components/AdminPanel';
import AdminPanelTable from '../../components/AdminPanelTable';

configure({ adapter: new Adapter() });

describe('<AdminPanel />', () => {
  let props = {
    users: [],
    match: {},
    count: 1,
    userDetails: {
      userrole: 'user'
    },
    history: {
      location: {
        search: ''
      }
    },
    limit: 1
  }
  const wrapper = shallow(<AdminPanel
    
    {...props} />
  )
  it('should render a <AdminPanel /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('it should render Action column if user is superadmin', () => {
    wrapper.setProps({
      ...wrapper.props,
      userDetails: {
        userrole: 'superadmin',
      }
      
    })
    expect(wrapper.find('.th-lg').at(6).text()).toBe('ACTION')
  });
  it('it should  not render the Action column if user is superadmin', () => {
    wrapper.setProps({
      ...wrapper.props,
      userDetails: {
        userrole: 'admin',
      }
      
    })
    expect(wrapper.find('.th-lg').at(6).text()).toBe('')
  });
  it('it should  not render <AdminPanelTable />', () => {
    wrapper.setProps({
      ...wrapper.props,
      userDetails: {
        userrole: 'admin',
      }
      
    })
    expect(wrapper.find('.th-lg').at(6).text()).toBe('')
  });
  it('it should  not render <AdminPanelTable />', () => {
    
    expect(wrapper.find('tbody')).toBeTruthy()
  });
  it('it should  not return null if users is undefined', () => {
    wrapper.setProps({
      ...wrapper.props,
      users: undefined
      
    })
    expect(wrapper.find('tbody').at(0).text()).toBeNull
  });
  it('it should render AdminPanelTable if users is not undefined', () => {
    wrapper.setProps({
      ...wrapper.props,
      users: ['egg', 'ball']
      
    })
    expect(wrapper.find('AdminPanelTable').exists()).toBeTruthy()
  });
});
