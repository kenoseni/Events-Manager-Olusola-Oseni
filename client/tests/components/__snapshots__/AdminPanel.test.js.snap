// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPanel /> should render a <AdminPanel /> component 1`] = `
<div
  className="container"
>
  <div
    className="row"
    id=""
  >
    <div
      className="col-xs-12 col-md-0"
    />
    <div
      className="col-xs-12 col-md-12"
      id="etable"
    >
      <div
        className="card w-40"
      >
        <div
          className="card-body"
        >
          <div
            className="table-wrapper"
          >
            <h3
              id="appusers"
            >
              Users
            </h3>
            <table
              className="table table-hover table-responsive"
            >
              <thead
                className="blue-grey text-white lighten-1"
              >
                <tr>
                  <th>
                    #
                  </th>
                  <th
                    className="th-lg"
                  >
                    ID
                  </th>
                  <th
                    className="th-lg"
                  >
                    FIRST NAME
                  </th>
                  <th
                    className="th-lg"
                  >
                    LAST NAME
                  </th>
                  <th
                    className="th-lg"
                  >
                    ISADMIN
                  </th>
                  <th
                    className="th-lg"
                  >
                    ROLE
                  </th>
                  <th
                    className="th-lg"
                  >
                    EMAIL
                  </th>
                </tr>
              </thead>
              <tbody
                id="adminpaneltable"
              />
            </table>
            <div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "space-between",
                  }
                }
              >
                <span
                  className="count"
                >
                  1 - 0 of 0
                </span>
                <span>
                  <Pages
                    count={1}
                    history={
                      Object {
                        "location": Object {
                          "search": "",
                        },
                      }
                    }
                    limit={1}
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-md-0"
    />
  </div>
</div>
`;
